lib/utils/hash_extension.rb

```diff
@@ -1,9 +1,7 @@
 module CoreExtensions
   refine Hash do
     def to_struct
-      OpenStruct.new(self.each_with_object({}) do |(key, val), acc|
-        acc[key] = val.is_a?(Hash) ? val.to_struct : val
-      end)
+      Struct.new(*self.keys).new(*self.values.map { |value| value.is_a?(Hash) ? value.to_struct : value })
     end
```

Comment:
I'm wondering if `to_struct` is even needed. It converts the notification passed to the [`group_started`](https://github.com/serpapi/turbo_tests/blob/703121054c785a364bf4ab499eeb72868e61bf68/lib/turbo_tests/reporter.rb#L53) method. It looks like, [`RSpec::Core::Notifications::GroupNotification`](https://github.com/rspec/rspec-core/blob/1e661db5c5b431c0ee88a383e8e3767f02dccbfe/lib/rspec/core/notifications.rb#L258) can be used instead.

@smileart Do you remember why [`RSpec::Core::Notifications::GroupNotification`](https://github.com/rspec/rspec-core/blob/1e661db5c5b431c0ee88a383e8e3767f02dccbfe/lib/rspec/core/notifications.rb#L258) was not used instead in https://github.com/serpapi/turbo_tests/pull/7?